### PR TITLE
fix: [IOBP-879,IOBP-880] Upgrade `react-native-vision-camera` to version `4.3.1`

### PR DESCRIPTION
**Pull request by Hantex9**
## Short description
This PR upgrades the `react-native-vision-camera` version to the v.`4.3.1` that fixes two of the bugs that are also reported in their issues.
1. [IOBP-879] The Camera is working fine on most devices. However, there are a few on which it just won't work. The camera will simply try to load and stay black. (https://github.com/mrousavy/react-native-vision-camera/issues/2716)
2. [IOBP-880] On some devices the camera image appears horizontally stretched (https://github.com/mrousavy/react-native-vision-camera/issues/2364)

## How to test
- Please test this PR on a real device to check that the camera is working correctly from the Scan section tab bar.

[IOBP-879]: https://pagopa.atlassian.net/browse/IOBP-879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[IOBP-880]: https://pagopa.atlassian.net/browse/IOBP-880?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1014,7 +1016,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   Alamofire: f36a35757af4587d8e4f4bfa223ad10be2422b8c
-  boost: 7dcd2de282d72e344012f7d6564d024930a6a440
+  boost: 9fa78656d705f55b1220151d997e57e2a3f2cde0
```
Comment (shadowsheep1) [inline ios/Podfile.lock:1019]: This shouldn't be changed. Am I right?

File: ios/Podfile.lock [hunk 2]
```diff
@@ -1046,7 +1048,7 @@ SPEC CHECKSUMS:
   pagopa-io-react-native-jwt: 662d4e722715996758b079774abea1996b057467
   pagopa-io-react-native-login-utils: a445c97d7d7dc86b089079029a1340f4c682128c
   pagopa-react-native-zendesk: 60a26f8a8072234789c34bb7c8a769c156eb57dc
-  RCT-Folly: 424b8c9a7a0b9ab2886ffe9c3b041ef628fd4fb1
+  RCT-Folly: 0080d0a6ebf2577475bda044aa59e2ca1f909cda
```
Comment (shadowsheep1) [inline ios/Podfile.lock:1051]: This shouldn't be changed. Am I right?

File: ios/Podfile.lock [hunk 3]
```diff
@@ -1142,4 +1144,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: c81fe8fa67af4bf6e27653958d177ac176ab0293
 
-COCOAPODS: 1.14.3
+COCOAPODS: 1.15.2
```
Comment (shadowsheep1) [inline ios/Podfile.lock:1147]: This shouldn't be changed. Am I right?
Be sure to run `bundle exec pod install`

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ios/Podfile.lock (hunk 2 above)
Comment (Hantex9) [reply ios/Podfile.lock]: You're right! I added the correct Podfile.lock by executing `bundle exec pod install` check it out from this commit: 96241092b9489cf386061c234653c4ee93cc14f9

File: ios/Podfile.lock (hunk 1 above)
Comment (Hantex9) [reply ios/Podfile.lock]: You're right! I added the correct Podfile.lock by executing `bundle exec pod install` check it out from this commit: 96241092b9489cf386061c234653c4ee93cc14f9

File: ios/Podfile.lock (hunk 3 above)
Comment (Hantex9) [reply ios/Podfile.lock]: You're right! I added the correct Podfile.lock by executing `bundle exec pod install` check it out from this commit: 96241092b9489cf386061c234653c4ee93cc14f9

**Review by shadowsheep1 — APPROVED** [review]
LGTM. 
Tested on both my Android and iOS test physical devices.

**Pull request merged**